uiowa/uiowa
# Now Spotlight article type; auto-tagging

Resolves #5562 




# How to test

Sync Iowa Now and make sure the site split's enabled:
```
git checkout now-spotlight && git pull
blt ds --site=now.uiowa.edu
drush @now.local en now_core
drush @now.local cim --source ../config/sites/now.uiowa.edu --partial
drush @now.local cim
drush @now.local cr
```

- Add some articles, including the newly added "spotlight" article type.
  - The Spotlight type should enable the "original publication date" field
- When an article is created, check that the associated tag is added
- Try different combos of including tags when creating an article, and check that a duplicate tag doesn't get added, that other tags aren't removed, etc.
- Edit an article and change the article type
  - After save, the new article type should be represented by a tag
  - After save, the previous article type should no longer be represented by a tag**
- Check that you can't delete tags associated with this auto-tagging (`achievements`, `featured`, `in-the-news`, `ui-spotlight`), but that you can delete other tags

** The current approach for this only removes tags from the previously selected article type, and not ALL other possible article types that may have been added manually. This allows a user to manually add additional tags that correspond to article types (eg 'feature', 'in-the-news', etc) in addition to the auto-tags and not lose them on save, while allowing the auto-tags to replace the old article type's tag with the new type's tag if it gets changed.
